java/gateway-core/src/main/java/com/avaulta/gateway/rules/Endpoint.java

```diff
@@ -55,6 +55,15 @@ public Optional<Set<String>> getAllowedMethods() {
         return Optional.ofNullable(allowedMethods);
     }
 
+    //if provided, headers provided will be pass-through to the endpoint
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    Set<String> supportedHeaders;
```

Comment:
> `allowed` may not be the best term, as unlike for `allowedQueryParams`, proxy will not return a 403 if request includes headers not in this list; it will just not forward them. But I think it's OK for consistency with `allowedQueryParams`, even if the logic isn't perfectly analogous.

Finally I've put `allowed` as part of the name, but I was between that and `supported`. Probably I'd have a better name in the future.